### PR TITLE
ci: setup pr github workflow

### PR DESCRIPTION
**Pull request by SaaiVenkat**
Combined the black, commit-message-lint, and the unit test workflow into one called `ci.yml`. This is beneficial because it could be reused in the release pipeline.

Resolves: #39

### CODE REVIEW

**Review by jooyoungseo — APPROVED** [review]
LGTM! Feel free to merge once ready.

**Comment by SaaiVenkat** [comment]
Professor @jooyoungseo . This PR looks like it includes the commits of #17 as well. It is because this PR has been branched off from that one, but still points to the `main` branch. I cannot point it to the `docs/add-quarto` branch because it is available only in my fork. So, PR #17 should be merged first. Once that is merged, this will show only the CI pipeline changes.

**Comment by jooyoungseo** [comment]
@SaaiVenkat Understood. Please feel free to go with that.

**Pull request merged**
